constraints/memory.air

```diff
@@ -2,13 +2,253 @@
 # MEMORY CONSTRAINTS MODULE
 ##########################################################################################
 #
-# Miden VM supports linear read-write random access memory. This memory is element-addressable,
-# meaning that a single value is located at each address, although reading and writing values
-# to/from memory in batches of four is supported and is optimized for.
+# The Memory chiplet provides linear read-write random access memory for the Miden VM.
+# Memory is element-addressable with addresses in range [0, 2^32), supporting both individual
+# element access and optimized word-aligned batch operations (4 elements per word).
 #
-# STATUS: Not implemented
+# MEMORY TABLE LAYOUT:
+# ┌─────────┬──────────────────────────────────────────────────────────────────────────┐
+# │ Column  │                               Purpose                                    │
+# ├─────────┼──────────────────────────────────────────────────────────────────────────┤
+# │   rw    │ Read/write selector: 1=read, 0=write                                     │
+# │   ew    │ Element/word access selector: 0=element, 1=word                          │
+# │   ctx   │ Context ID for execution context separation                              │
+# │ w_addr  │ Word address (must be divisible by 4 for aligned access)                 │
+# │ idx0/1  │ Element index within word [0,3] - binary decomposition of low 2 bits     │
+# │   clk   │ Clock cycle when operation occurred                                      │
+# │ v0-v3   │ Four field elements stored in memory word                                │
+# │ d0/d1   │ Delta tracking columns for monotonicity verification                     │
+# │  d_inv  │ Inverse delta column for consecutive transitions                         │
+# │ f_scw   │ Same context/word address flag for sequential access optimization        │
+# └─────────┴──────────────────────────────────────────────────────────────────────────┘
+#
+# STATUS: Partially implemented (missing bus integration)
```

Comment:
The documentation mentions that memory constraints are missing bus integration with constraint: bchip' = bchip*vmem (degree 5). Should we add a placeholder for this bus integration constraint or mention it more explicitly in the status?